src/ccall.cpp

```diff
@@ -2092,8 +2092,11 @@ jl_cgval_t function_sig_t::emit_a_ccall(
     }
     else if (symarg.fptr != NULL) {
         ++LiteralCCalls;
-        Type *funcptype = PointerType::get(functype, 0);
-        llvmf = literal_static_pointer_val((void*)(uintptr_t)symarg.fptr, funcptype);
+        Type *funcptype = functype->getPointerTo(0);
+        auto gv = julia_pgv(ctx, "ccall_fptr#", (void*)(uintptr_t)symarg.fptr);
+        auto casted = emit_bitcast(ctx, gv, funcptype->getPointerTo(0));
+        llvmf = ctx.builder.CreateAlignedLoad(funcptype, casted, Align(sizeof(void*)));
```

Comment:
This is not worth changing, since it is bad user code already anyways
```suggestion
        Type *funcptype = PointerType::get(functype, 0);
         llvmf = literal_static_pointer_val((void*)(uintptr_t)symarg.fptr, funcptype);
```

Reply:
Otherwise, LGTM